### ISSUE TITLE
bottom nav toolbar appears when viewing a offline list

### ISSUE THREAD

**Issue by ztNFny** [cgeo/cgeo#12443]
### Describe your problem!

View a stored list
Switch to map view
-> bottom nav toolbar is visible and "live map" is selected
This is a recent regression

### How to reproduce?

.

### Reproducible

Yes

### c:geo Version

2021-12-29-NB

**Comment by fm-sys** [comment]
I would call this "intended behavior". Why should we need a bottom nav if we never show it? ;-)

**Comment by ztNFny** [comment]
I didn't complain about bottom nav so far because i never hot to see it as it didn't show up when offline caching.

Besides the general points about bottom nav this introduces at least a few concrete points:
For one it's not a "live map"
Then that change also broke the "top-chip" back behaviour when navigating to a cache. 
And now going from list to map i get a map of the caches i was viewing but the reverse isnt true.

**Comment by Lineflyer** [comment]
I can confirm the bottom nav is visible (not sure if its a recent regression) but the map is not in live mode for me (which would really be wrong).

**Comment by moving-bits** [comment]
To my understanding we had to different ways to display a map, after having selected a stored list:
- If you tap the bottom nav "live map" button bottom nav should be visible, and map should be in "live" mode (if enabled), otherwise show all stored caches in the selected viewport
- If instead you push the toolbar "show on map" button, it should display the caches of the current list, and no bottom nav (at least it has been this way before, but isn't any longer)

**Comment by ztNFny** [comment]
The intention of bottom navigation is described is to switch between different parts of the application. Switching between map and listview is definitely not different part of the application
- offline list vs lifemap ok come that's separate things, but not viewing a listing different modes

**Comment by Lineflyer** [comment]
I would also favor, that the bottom nav is not shown while staying "inside" of one main activity.  Otherwise we could also start to discuss whether to show it always (e.g. while in cache details). That IMHO does not make sense.

**Comment by ztNFny** [comment]
If bottom nav were to be expanded than really a lot more thought then "let's replace dashboard" (which actually never happened se still have to dashboard just less functional) needs to go into it.
I made some suggestions for a big redesign with a different concept in https://github.com/cgeo/cgeo/issues/11841#issuecomment-945124765 back when bottom back was first discussed

**Comment by Lineflyer** [comment]
FYI. The commit introducing this change is d9e875ea6a5e0fb893f3cc15424d67d8e0d462f6

**Comment by ztNFny** [comment]
Found another issue caused by this: navigating to a cave the wrong menu shows

**Comment by fm-sys** [comment]
>I didn't complain about bottom nav so far because i never hot to see it as it didn't show up when offline caching.

@ztNFny  Somehow this statement looks strange to me, cause 3 months ago you had mentioned somewhere, that you also did not like the old dashboard?  So what *do* you like? 

Anyway, let's start at the beginning:

### Intended Behavior for a bottom nav (according to [material](https://material.io/components/bottom-navigation#behavior)) 
> On Android: the app navigates to a destination’s top-level screen. Any prior user interactions and temporary screen states are reset, such as scroll position, tab selection, and in-line search.
When moving downward in the app’s hierarchy (from a parent screen to a child screen), a bottom navigation bar **can** be displayed persistently for quick navigation between an app’s sections. 

The "big question" we need to discuss is this "can".

A bottom navigation (or any other navigation element) should be designed to make navigation inside the app as fast as possible.

Imagine a normal use case (al least for me):
Open a list or live map --> open a cache --> open compass view --> open map for better navigation

As soon as I have found the cache, I log the cache via the cache popup and want to search the next cache. How would this work with different solutions? 

Currently (with the "new" behavior) you can directly switch to "lists" or reselect "map" to come to the live map. One click and you are there. 

If the bottom navigation would not be shown, you would need 3 (!) back clicks. If the user had switched multiple times between map and compass even 5, 7 or even more clicks would be necessary. 

### The main problem are our hierarchies... 

But no matter if we should the bottom navigation for offline-list-maps or not: This discussion only hides the general fact that cgeo is way too complex in its structure. Especially Map and CacheListActivity are used in too many totally different ways... 🙄

If we find a different way to reduce the app hierarchy to ideally 1-2 child activities max, I would be more than fine with that solution. 

But until this goal is reached, showing the bottom navigation everywhere (=for all map contexts) seems like a good solution to me. 

An idea to simply the hierarchy would e.g. be the following: Have only one activity for compass and map preview. Don't know if this will work in real life in the way I imagine it, but would this be something you would support? Although this would require to create a good abstraction layer first, to share the map implementation as so-called Fragment between both activities...

![IMG_20220101_205733_copy_216x451](https://user-images.githubusercontent.com/64581222/147859162-bb2a5e5c-520f-4389-9565-d3710d67e25c.jpg)

**Comment by ztNFny** [comment]
> @ztNFny Somehow this statement looks strange to me, cause 3 months ago you had mentioned somewhere, that you also did not like the old dashboard? So what do you like?

Not liking one bad solution means I have to like the next bad solution?
The current bottom nav is 1:1 dashboard, just turned from a sometimes-visibile-fullscreen to an always-visibile-part of the screen.

When I open maps view from a list bottom nav currently says I'm in "Live Map" activity. That statement is 100% false!
I'm in a map activity, but most definitly not the "Live map" one. Click the already active bottom nav element and you'll see that the map then displays something completly different - this is just wrong! 

> Imagine a normal use case (al least for me): Open a list or live map --> open a cache --> open compass view --> open map for better navigation
> 
> As soon as I have found the cache, I log the cache via the cache popup and want to search the next cache. How would this work with different solutions?
> 
> Currently (with the "new" behavior) you can directly switch to "lists" or reselect "map" to come to the live map. One click and you are there.

For me a normal use is "Open map, select cache, in popup select the 'compass' activity which is set to Map (most people I know use map, never compass)". Then going back to the map to select the next cache is exactly 1 back button press.

> Especially Map and CacheListActivity are used in too many totally different ways... roll_eyes
> If we find a different way to reduce the app hierarchy to ideally 1-2 child activities max, I would be more than fine with that solution.

I proposed a concept to achieve just that in https://github.com/cgeo/cgeo/issues/11841#issuecomment-945124765

In short: c:geo has 2 main views, list and map. Everything around that is just different modes of displaying those (like live, offline, ...)

> Have only one activity for compass and map preview. Don't know if this will work in real life in the way I imagine it, but would this be something you would support

If you ask me personally - compass can be removed completly, I see not a single usecase for it. Map view is better in all regards.

**Comment by fm-sys** [comment]
Another thing to consider in the same context is the "show as list" feature when being on the map. Just to throw that in...

**Comment by fm-sys** [comment]
>When I open maps view from a list bottom nav currently says I'm in "Live Map" activity. That statement is 100% false!

If it's only about which element is hi we could keep "List" selected. BTW, that would even be aligned with the way how "show as list" is currently implemented...

---

Oh, and torward you alternative idea. That would be a 180° turn for sure, but here is the ToDo list:
a) it should be intuitive for users 
b) we need a minority for it in the team 
c) someone needs to implement it

Don't get me wrong, I definitely don't want to block your idea and think that it might work as well. But at least until now, absolutely no one has reacted to it. Neither in a positive nor negative way...

**Comment by ztNFny** [comment]
> If it's only about which element is hi we could keep "List" selected.

That would at least be more correct. More issues were mentioned in https://github.com/cgeo/cgeo/issues/12443#issuecomment-1003345300

> Don't get me wrong, I definitely don't want to block your idea and think that it might work as well. But at least until now, absolutely no one has reacted to it. Neither in a positive nor negative way...

I'm not saying the concept I proposed there should be implemented, in fact I see a lot of detail questions that are unsolved in there.

But I see just as many detail problems with the current implementation, the major ones being:
- a toolbar that provides permanent access to "start" items when it's unlikely that people actually switch between those sub-activities during the use of the app. The only one of those 5 items in bottom nav I ever use is "Stored", why do I have to see 4 items I never use at all times!? Depending on the way you use c:geo your main activity may be a different one, but I doubt you'll find anyone permanently switching between those. As such the 5 current items aren't suited to make up a bottom nav
- a dashboard view that's completly wasted, it should be removed completly

Sorry to say it that harshly, but with the current implenetation I'm completely missing any idea of what it's trying to achieve / solve. I like the concept of bottom nav itself, but the current implementation doesn't make any sense to me.

Comparing the old (bad) dashboard solution to the new bottom nav I actually see more problems with bottom nav than I saw with dashboard. It feels like a change for the sake of change.
The old dashboard was ugly and outdated, but now we're trading that for a not-thought-through concept of using that app that will force users to relearn a lot of workflows - which can be worth it if there's some advantage - but I'm not seeing anything that actually gets easier with bottom nav.
Yes, I can now switch between stored and nearby with 1 click instead of 2, but even that is something that will very seldomly happen.

**Comment by ztNFny** [comment]
The issue of "old dashboard is ugly and is just an intermediate page the user needs to pass to get to where he actually wants" could be address with:
- put status info in a sidebar menu
- let the user select a default start activity

From that point on provide an easy way to switch between list and map views. And that's already 90% of what you need during normal use of the app.

**Comment by Lineflyer** [comment]
> The "big question" we need to discuss is this "can".

Without understanding all technical aspects about the stacks, fragments, etc, my answer/preference to that question would be to NOT show it in all those cases.
The main reason for me is, that I don`t like to use the screen space on all screens. I rather accept to press "back" until I reach it again.

**Comment by moving-bits** [comment]
Have not been able to follow the discussion in every detail, but want to broaden the view on how people use c:geo:

If there's one single observation we have made throughout the last years, then it's that people are using c:geo in very different ways. This sort of forbids things like "nobody uses xy" or "everyone only needs a and b", IMHO, and we should be very careful not to impose our own way of using c:geo to all other users.

If we change something in this regard, especially removing core functionality or making something harder to access, for whatever reason, it should be a careful decision, and a majority of the team should agree to such a change.

**Comment by ztNFny** [comment]
> If there's one single observation we have made throughout the last years, then it's that people are using c:geo in very different ways. 

I think that's clear to everyone in this discussion.

And that "different ways - side by side" is something that worked good with the old dashboard - ugly and outdated as it might be - as you only saw "the other ways" for a brief moment when starting the app. If you don't use Stored, Live, Nearby, .. then during use of cgeo you were never confronted with them.
But now they are always visible in bottom nav, so now suddenly "other peoples ways of using cgeo" are getting in your way. Having to ignore 4 always-there menu items is a whole different thing then ignoring 4 dashboard buttons you see one a day.

**Comment by moving-bits** [comment]
Having said that, _my_ personal use of the old mainscreen functions is:
- frequently using "live map" (which never was only a "live map") and "stored"
- sometimes using "search"
- (nearly) never used "nearby" - having map for this is much more useful
- never used "go to" or the global single cache type filter (which meanwhile does not exist anymore)
- also I'm frequently using some of the mainscreen's menu functions, e. g. "PQ / bookmark lists", "settings" and some others

In contrast to other usage patters described here, I'm frequently switching between cache list, cache list map, global map and settings, and I'm frequently getting annoyed by having to "crawl up the ladder for so many steps, just to walk it down on the other side, and then back again the same tedious procedure". For me, the new bottom navigation in its current state is not perfect, but already a great relief!

If I were to have a few small wishes for free (which leaves out some possible more radical changes, though there may be some valid options here), I would vote for:
- keep bottom navigation as is now, but fix a few things like the one issue described at the top of this thread or #12423
- optionally remove the "nearby" button (but it doesn't harm too much)
- don't show bottom navigation when mapping a cache list (so you have an easy to spot visible indication that this is a different type of map)
- try to remember more states (#11926) (although this might need some more detailed discussion which path to pursue)

Just my 2ct :-)

**Comment by moving-bits** [comment]
> Having to ignore 4 always-there menu items is a whole different thing then ignoring 4 dashboard buttons you see one a day.

Here I disagree. Having all this ping-pong of pushing back several times to come back to mainscreen just to walk down the next hierarchy, and then all the way back, is much more hassle than ignoring the lowest 1.5 cm of my phone's screen. Especially if you have to do it several times during a caching tour.
Bottom navigation is not taking up much space, at least on today's 16:9 => 21:9 (or whatever shaped) phones, and the benefits are immense. - at least for people which frequently switch between the main activities.

**Comment by ztNFny** [comment]
> Especially if you have to do it several times during a caching tour.

I wonder, what exactly do you climb so often for? You mentioned settings above, maybe there's something in there that should be more easily available (from map} if you change it so often?
For me the only item in bottom nav i'd sometimes use is live map - but switching live on/off there's already a toggle in maps main toolbar.

When I was out caching yesterday I accidentally clicked on live map instead of map settings and Nearby instead of zoom buttons on the map several times - very annoying experience.
Those buttons used to be at the side of the screen, very easy to hit. Now they're somewhere in the center. This con for me outweighs any saved back clicks by far.

**Comment by SammysHP** [comment]
I mostly stay in "map of list" where I want to see only caches of that list. The I open "cache details" frequently, sometimes starting "compass navigation", very rarely "show cache on map" (mostly for multi caches with waypoints). In all those situations I never need the bottom navigation.

**Comment by fm-sys** [comment]
IMHO this whole discussion here doesn't bring any value. May too much off-topic stuff and comments like "too long, didn't read". Let's postpone this discussion till next Sunday when we have our next dev meeting...

**Comment by Lineflyer** [comment]
The following has been decided today in our dev call regarding when to show/not show the bottom nav:
- Stored - Show list on map = No bottom nav
- Stored - Shows list on map + Users enables live mode = Show bottom nav (live map icon active)
- Stored - Open cache - Show map for cache = No bottom nav
- Stored - Open cache - Show map for cache + User enables live mode= Show bottom nav (live map icon active)
- Live Map - Show as list = No bottom nav
- Search - Search result = Show bottom nav
- Nearby search (in fact same case as above) = Show bottom nav

For those not participating in the call:
- This has been discussed case by case taking into account how user might understand it and trying to balance the presence of bottom nav to places where it is potentially understandable to be shown and potentially even needed. 
- This describes the starting point, which was seen as "Better less shown than to much". We can more easily add it to some more places later on (based on beta or first release feedback)  than remove it again in places where it was already used.
- Technical feasibility and potential side effects needs to be checked though

**Comment by fm-sys** [comment]
PR #12513 will solve this issue...
